phoenix-core-server/src/main/java/org/apache/phoenix/coprocessor/MetaDataEndpointImpl.java

```diff
@@ -3904,6 +3904,13 @@ private MetaDataMutationResult dropIndexes(RegionCoprocessorEnvironment env, PTa
                 // that a change in index state doesn't
                 // occur while we're dropping it.
                 acquireLock(region, indexKey, locks);
+                // invalidate server metadata cache when dropping index
+                List<InvalidateServerMetadataCacheRequest> requests = new ArrayList<>();
+                requests.add(new InvalidateServerMetadataCacheRequest(tenantId,
```

Comment:
We are invalidating server cache for table in MDEI#mutateColumn [here](https://github.com/apache/phoenix/blob/PHOENIX-6883-feature/phoenix-core-server/src/main/java/org/apache/phoenix/coprocessor/MetaDataEndpointImpl.java#L3296-L3299).

@palashc  Do you mean to invalidate the indexes?

Reply:
@shahrs87 Yes, we need to invalidate the entry for any indexes which we are dropping here. Are you suggesting I change the comment here?

Reply:
@shahrs87 Pushed the change as discussed offline.